Add MMonit Statuses

Jumba opened the pull request.

Commit:
Add support for reading service status from M/Monit

Commit:
Fix broken initializations

Commit:
Process comments and add scheduler for mmonit status

lmeijvogel commented on status/server.rb:
```diff
@@ -4,16 +4,13 @@
   def led_color
-    LEDColor.hex(led)
+    conversion_hash = { 1 => :error, 2 => :warning, 3 => :ok }
+    LEDColor.hex(conversion_hash.include?(led) ? conversion_hash[led] : :disabled)
```
`conversion_hash.fetch(led, :disabled)` zou ook kunnen.

lmeijvogel commented on m_monit_adapter.rb:
```diff
@@ -0,0 +1,38 @@
+    if mmonit
+      mmonit.disconnect
+    end
+  end
```
Deze roep je nu zo aan:
```ruby
adapter.connect
@servers = adapter.statuses
adapter.disconnect
```
Maar als er iets misgaat in `adapter.statuses`, dan wordt `#disconnect` nooit aangeroepen.

Je kunt in de aanroepende code een `ensure` toevoegen:
```ruby
begin
  adapter.connect
  @servers = adapter.statuses
ensure
  adapter.disconnect
end
```
maar je kunt hem ook hier aanpakken door zoiets te doen:
```ruby
class MMonitAdapter
  def connect
    @mmonit ||= MMonit::Connection.new(
      ssl: false,
      username: @username,
      password: @password,
      address: @address,
      port: @port
    )
  
    @mmonit.connect
  
    yield self
  ensure
    if @mmonit
      @mmonit.disconnect
      @mmonit = nil
    end
  end
end
```

lmeijvogel commented on app.rb:
```diff
@@ -1,83 +1,116 @@
-  res = Net::HTTP.start(uri.hostname, uri.port) do |http|
-    http.request(req)
+    settings.scheduler.every '5s', first: :now do
+      App.load_servers
```
Voor de zekerheid: Weet je zeker dat `@servers` persisted is tussen requests? Zo nee, dan zou je misschien beter bijv. een simpele cache in Redis kunnen doen:
```ruby
redis.hsetall("mmonit_server_statuses", statuses)
redis.expire("mmonit_server_statuses", 5)
```

en die laten uitlezen of instellen door de webapp. Vraag is of het nodig is: Is mmonit zo langzaam dat periodiek ophalen nodig is of kan het ook tijdens een request gedaan worden?

lmeijvogel commented on status/server.rb:
```diff
@@ -0,0 +1,31 @@
+
+  # If the LED is green all services are okay. Check this instead of looping through all services.
+  def ok?
+    led == 2
```
In de `conversion_hash` hierboven lijkt `3` de waarde voor `:ok` te zijn, klopt dat?

lmeijvogel commented on app.rb:
```diff
@@ -1,83 +1,116 @@
-end
+  def read_mmonit_status
+    if App.servers
+      App.servers.map do |server|
```
Je kunt hier nog `Array(App.servers)` doen, dan hoeft de `if App.servers` niet: `Array(nil) == []`.

lmeijvogel reviewed (commented):
Gedane wijzigingen zijn al prima, zijn nu vooral wat puntjes op de i :D

Commit:
Add Rack config and deploy configuration

Updates

Updates

update

Update the ruby version to 2.3.3 since 2.3.1 isn't installed on staging

Use https instead of ssh

Commit the lock file

Change passenger config

Test basic configuration

Change the staging configuration

Change the way the servers list is accessed

Update deploy configuration

Create bundle folder in shared path

Get server status on request
